Riot/Modules/Contacts/DataSources/ContactsDataSource.m

```diff
@@ -196,21 +196,24 @@ - (void)searchWithPattern:(NSString *)searchText forceReset:(BOOL)forceRefresh h
                 [hsUserDirectoryOperation cancel];
                 hsUserDirectoryOperation = nil;
             }
+            
+            MXWeakify(self);
```

Comment:
Is weakifying and strongifying the right thing to do in this case as the request could have a longer lifetime than the data source?